content/glossary/build_tool.md

```diff
@@ -53,4 +61,18 @@ flowchart LR
     compiler --> .pyc
 ```
 
-The totality of ancestors for a given artifact may be represented as an [artifact tree](/glossary/artifact_tree).
\ No newline at end of file
+The totality of ancestors for a given artifact may be represented as an [artifact tree](/glossary/artifact_tree).
+
+## Code Generators
+
+Typically, source code files are hand written by humans, and as such are [leaf artifacts](http://localhost:1313/glossary/artifact/#leaf-artifacts) in the [artifact tree](/glossary/artifact_tree).
+
+Sometimes source code files are themselves **generated** by a code generator from inputs.
+
+```mermaid
+flowchart LR
+    codegeninput[codegen input] --> codegenerator[[code generator]] --> generatedsrc[generated source code file]
```

Comment:
Took a minute to parse the semantics behind this. It's probably more important to simply highlight that this is an 'input', rather than to label it a 'codegen input'. Suggest deleting 'codegen'.